### PR TITLE
Update KetlCred and add tests

### PR DESCRIPTION
**Pull request by wilsonplau**
Merged all KetlCred-related PRs into one:
- Grants KetlCred to author instead of sender (https://trello.com/c/sXd3Ux0d/3276-fix-ketlcred-contract-to-give-cred-to-whoever-is-being-upvoted-not-to-whoever-upvotes)
- Users should not be able to get KetlCred for upvoting their own posts (https://trello.com/c/9HJ5hUzm/3274-disalow-giving-karma-to-self)
- KetlCred should only be granted if the Reaction is an upvote (https://trello.com/c/8JtFZiHn/3277-only-give-ketl-cred-when-upvoting)
- Add tests for KetlCred features (https://trello.com/c/VDsEbHOj/3342-add-tests-for-ketlcred)

### CODE REVIEW

File: contracts/OBSSStorage.sol [hunk 1]
```diff
@@ -198,6 +198,10 @@ contract OBSSStorage is OwnableUpgradeable, ERC2771Recipient {
   // KetlCred
 
   function grantKetlCred(AddReactionRequest memory reactionRequest) internal {
+    Post memory post = feeds.getPost(
```
Comment (backmeupplz) [inline contracts/OBSSStorage.sol:201]: I wonder if we do this in `feed.addReaction` too? Maybe we should pass the `post` to `feed.addReaction`?

File: contracts/superclasses/Posts.sol [hunk 2]
```diff
@@ -130,6 +130,12 @@ contract Posts is KetlGuarded {
   }
 
   // Posts
+  function getPost(
```
Comment (backmeupplz) [inline contracts/superclasses/Posts.sol:133]: Hm, do we absolutely need this accessor? `posts` is public after all

Comment (wilsonplau) [reply contracts/superclasses/Posts.sol:133]: I think it's possible, but it doesn't pass it as a `Post` struct, but like a raw array. Using a function allows it to be returned as a `Post`

Comment (backmeupplz) [reply contracts/superclasses/Posts.sol:133]: Just checked it, for some reason `feeds.posts[x][y]` doesn't return `Post` 😅 adding `getPost` is fine then

File: contracts/OBSSStorage.sol (hunk 1 above)
Comment (wilsonplau) [reply contracts/OBSSStorage.sol:201]: We don't fetch posts anywhere else. I did think about moving this to `addFeedReaction` and only calling `grantKetlCred` if the reaction was an upvote, but that logic would be repeated in `addProfileReaction`, so I opted to keep the logic in `grantKetlCred` instead.

**Review by backmeupplz — APPROVED** [review]

**Pull request merged**